pkg/kube/client.go

```diff
@@ -433,17 +433,25 @@ func perform(infos Result, fn ResourceActorFunc) error {
 }
 
 func createResource(info *resource.Info) error {
-	// create resource with additional label of the force-upgrade option
+	obj, err := resource.NewHelper(info.Client, info.Mapping).Create(info.Namespace, true, info.Object)
```

Comment:
Originally, `createResource` is called on both creation and update. Now on creation this is done with no label injection. Is this an expected change?

Reply:
yes, and label injection will be added in the https://github.com/rancher/helm/blob/3b020290fec3c8ef69a4b5643b083b7182646869/pkg/kube/client.go#L102 to create and update. 
and for those recreated object, it will call the `recreateResource`  function precisely, and the `io.cattle.field/appId` will be added accordingly.